solr/core/src/java/org/apache/solr/handler/admin/CoreAdminOperation.java

```diff
@@ -919,7 +928,93 @@ public void call(CallInfo callInfo) throws Exception {
         }
       }
     }
+  },
+  CREATESNAPSHOT_OP(CREATESNAPSHOT) {
+    @Override
+    public void call(CallInfo callInfo) throws Exception {
+      CoreContainer cc = callInfo.handler.getCoreContainer();
+      final SolrParams params = callInfo.req.getParams();
+
+      String commitName = params.required().get(CoreAdminParams.COMMIT_NAME);
+      String cname = params.required().get(CoreAdminParams.CORE);
+      try (SolrCore core = cc.getCore(cname)) {
+        if (core == null) {
+          throw new SolrException(ErrorCode.SERVER_ERROR, "Unable to locate core " + cname);
+        }
+        if (core.getDeletionPolicy().getLatestCommit() == null) {
```

Comment:
Firstly, this check can go below after we get it on line 949; why get it twice.  Secondly... I wonder if it's null in an initial empty-index case?  If it is, this error message is wrong/confusing... and perhaps we can support that or no?